rust/operator-binary/src/catalog.rs

```diff
@@ -0,0 +1,243 @@
+        property: impl Into<String>,
+        file_name: impl Into<String>,
+    ) {
+        let property = property.into();
+        let env_name = calculate_env_name(&self.name, &property);
+        self.add_property(&property, format!("${{ENV:{env_name}}}"));
+        self.load_env_from_files.insert(env_name, file_name.into());
+    }
+
+    fn add_configmap_property(
+        &mut self,
+        property: impl Into<String>,
+        config_map: impl Into<String>,
+        config_map_key: impl Into<String>,
+    ) {
+        let property = property.into();
+        let env_name = calculate_env_name(&self.name, &property);
+        self.add_env_property(
+            &property,
+            EnvVar {
+                name: env_name,
+                value: None,
+                value_from: Some(EnvVarSource {
+                    config_map_key_ref: Some(ConfigMapKeySelector {
+                        name: Some(config_map.into()),
+                        key: config_map_key.into(),
+                        ..ConfigMapKeySelector::default()
+                    }),
+                    ..EnvVarSource::default()
+                }),
+            },
+        );
+    }
+}
+
+fn calculate_env_name(catalog: impl Into<String>, property: impl Into<String>) -> String {
+    let catalog = catalog.into().replace('.', "_").replace('-', "_");
+    let property = property.into().replace('.', "_").replace('-', "_");
+    format!("CATALOG_{catalog}_{property}").to_uppercase()
+}
+
+impl CatalogConfig {
+    pub async fn from_catalog(
+        catalog: TrinoCatalog,
+        client: &Client,
+    ) -> Result<CatalogConfig, FromTrinoCatalogError> {
+        let catalog_name = catalog.name();
+        let mut config = CatalogConfig {
+            name: catalog_name.to_string(),
+            properties: BTreeMap::new(),
+            env_bindings: Vec::new(),
+            load_env_from_files: BTreeMap::new(),
+            init_container_extra_start_commands: Vec::new(),
+            volumes: Vec::new(),
+            volume_mounts: Vec::new(),
+        };
+
+        config.add_property("connector.name", catalog.spec.connector.name());
+
+        match &catalog.spec.connector {
+            stackable_trino_crd::catalog::TrinoCatalogConnector::Hive(hive) => {
+                config.add_configmap_property(
+                    "hive.metastore.uri",
+                    hive.metastore.config_map.clone(),
+                    "HIVE",
+                );
+                // No authorization checks are enforced at the catalog level.
+                // We don't want the hive connector to prevent users from dropping tables.
+                // We also don't want that the hive connector makes decisions on which user is allowed to do what.
+                // This decision should be done globally (for all catalogs) by OPA.
+                // See https://trino.io/docs/current/connector/hive-security.html
+                config.add_property("hive.security", "allow-all");
```

Comment:
I would like to keep this stuff in the operator-binary and not move it to crd (Is this correct?).
What do you think about an static function within operator-binary/src/catalog.rs?

Reply:
Im thinking a little in terms of getting rid of huge files and better unit test ability. I think this belongs to wherever the connector is located (so in crd). 
We could make a catalog module (in crd) and use on file per connector which is then just imported and used in the catalog.rs.
For now its ok because there is only one connector. 
If you think this is out of scope im ok with refactoring later, but i would prefer having this separated cleanly from the beginning?